src/main/java/seedu/address/logic/commands/UnmarkCommand.java

```diff
@@ -0,0 +1,135 @@
+
+    /**
+     * @param markPersonDescriptor details to edit the person with
+     */
+    public UnmarkCommand(MarkPersonDescriptor markPersonDescriptor) {
+        requireNonNull(markPersonDescriptor);
+
+        this.markPersonDescriptor = new MarkPersonDescriptor(markPersonDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        Index index = Index.fromOneBased(1);
+        requireNonNull(model);
+        if (!model.isFullView() || model.isDayView()) {
+            throw new CommandException(MESSAGE_NOT_VIEW_MODE);
+        }
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToUnmark = lastShownList.get(index.getZeroBased());
+        Person unmarkedPerson = createUnmarkedPerson(personToUnmark, markPersonDescriptor);
+
+        model.setPerson(personToUnmark, unmarkedPerson);
+        return new CommandResult(String.format(MESSAGE_UNMARKED_PERSON_SUCCESS, unmarkedPerson));
+    }
+
+    private static Person createUnmarkedPerson(Person personToUnmark, MarkPersonDescriptor markPersonDescriptor)
+            throws CommandException {
+        Name name = personToUnmark.getName();
+        Phone phone = personToUnmark.getPhone();
+        LessonPlan lessonPlan = personToUnmark.getLessonPlan();
+        HomeworkList updatedHomeworkList = getUpdatedHomeworkList(personToUnmark, markPersonDescriptor);
+        AttendanceList updatedAttendanceList = getUpdatedAttendanceList(personToUnmark, markPersonDescriptor);
+        SessionList sessionList = personToUnmark.getSessionList();
+        GradeProgressList gradeProgressList = personToUnmark.getGradeProgressList();
+        Set<Tag> tags = personToUnmark.getTags();
+
+        return new Person(name, phone, lessonPlan, updatedHomeworkList,
+                updatedAttendanceList, sessionList, gradeProgressList, tags);
+    }
+
+    public static HomeworkList getUpdatedHomeworkList(Person personToUnmark, MarkPersonDescriptor markPersonDescriptor)
+            throws CommandException {
+        HomeworkList updatedHomeworkList = personToUnmark.getHomeworkList();
+        Optional<Index> homeworkIndex = markPersonDescriptor.getHomeworkIndex();
+        if (homeworkIndex.isEmpty()) {
+            return updatedHomeworkList;
+        }
+        if (!updatedHomeworkList.isValidIndex(homeworkIndex.get())) {
+            throw new CommandException(HomeworkList.MESSAGE_INVALID_HOMEWORK_INDEX);
+        }
+        updatedHomeworkList.unmarkAtIndex(homeworkIndex.get());
+        return updatedHomeworkList;
+    }
+
+    public static AttendanceList getUpdatedAttendanceList(Person personToUnmark,
+                                                          MarkPersonDescriptor mPerDesc) throws CommandException {
```

Comment:
Indentation

Also`getUpdatedHomeworkList` spells out markPersonDescriptor in full so not sure why it's shortened to mPerDesc here?

Reply:
initially was too long, will address this in the next subsequent commit